### PR TITLE
Root NS record support

### PR DESCRIPTION
**Pull request by acm1**
Refactored this a bit, cleaned up and added tests, and merged the latest master branch. I'm opening a separate PR against master as well to test mergeability.

### CODE REVIEW

File: octodns/zone.py [hunk 1]
```diff
@@ -140,6 +140,11 @@ def changes(self, desired, target):
                                record)
                 changes.append(Delete(record))
             else:
+                if not target.supports(record):
+                    self.log.debug('changes:  skipping record=%s %s - %s does '
+                                   'not support it', record.fqdn, record._type,
+                                   target.id)
+                    continue
```
Comment (acm1) [inline octodns/zone.py:147]: This prevents changes that are unsupported in the target provider from being calculated. Since the `supports` method on `BaseProvider` now checks that root NS record management is supported and enabled, this will filter out changes to root NS records unless actually desired and supported. This also makes all the existing tests pass since none of the behavior changes. I've added some new tests to check the `manage_root_ns` functionality as well.

File: octodns/provider/base.py [hunk 2]
```diff
@@ -111,3 +100,9 @@ def apply(self, plan):
     def _apply(self, plan):
         raise NotImplementedError('Abstract base class, _apply method '
                                   'missing')
+
+    def supports(self, record):
+        if record.name == "" and record._type == "NS":
+            return self.SUPPORTS_ROOT_NS and self.manage_root_ns
```
Comment (acm1) [inline octodns/provider/base.py:106]: This replaces the root NS filtering logic in `_check_root_ns`.

**Comment by joschi36** [comment]
Thanks for helping me here. This PR already got a little dusty, and I was working at other stuff.
I will review it together with my colleague @josvo.
We can't say much about the quality of the code, but we can check the function with our setup :smile:
